src/api/mod.rs

```diff
@@ -675,9 +674,9 @@ impl<'a> AuthenticatedApi<'a> {
                 }
             }
 
-            let pagination = resp.pagination();
+            let next = resp.next_pagination_cursor();
             rv.extend(resp.convert::<Vec<Artifact>>()?);
-            if let Some(next) = pagination.into_next_cursor() {
+            if let Some(next) = next {
```

Comment:
Is there a reason the `extend` is sandwitched between getting the `next` and matching it?
It might be simpler to read having the call and the the match on the same line.

Reply:
Yes, resp.convert consumes ownership of resp